plugins/modules/win_dns_record.ps1

```diff
@@ -8,10 +8,13 @@
 $spec = @{
     options = @{
         name = @{ type = "str"; required = $true }
+        port = @{ type = "int"; default = [int]"" }
```

Comment:
Yes, `-Port` `-Priority` and `-Weight` is required parameters that's why I set their default values as 0,0 and an empty string for the port because Port 0 is invalid. If users don't want to mention weigh, priority, or port to their record then also it should work with default values.

Reply:
You want to enforce the user to specify these when `type: SRV` so while they aren't required in all cases they should be required in that case. Setting the default values like it is in the PR currently is not a good idea, especially because a port of 0 doesn't make any sense. The suggestion is to remove the defaults and use that `required_if` value to have it mandatory when `type: SRV` but not in other cases.